File: 金辉/H.java
package testBUPT;

import java.util.Scanner;

public class H {
	public static void main(String[] args) {
		Scanner cin = new Scanner(System.in);
		int count = cin.nextInt();
		for(int i=0;i<count;i++) {
			int a = cin.nextInt();
			int b = cin.nextInt();
			if(helper(a) == b && helper(b) == a) 
				System.out.println("YES");
			else 
				System.out.println("NO");
			
		}
	}
	public static int helper(int n) {
		int res = 0;
		for(int i=2;i<n;i++) {
			if(n%i ==0) res += i;
		}
		return res + 1;
	}
}

File: 李尚洋/H.cpp
#include<iostream>
using namespace std;

int check(int a){
	int sum=0;
	for(int i=1;i<=a/2;i++){
		if(a%i==0){
			sum+=i;
		} 
	}
	return sum;
}

int main(){
	
	int a,b;
	int n;
	while(cin>>n){
		while(n--){
			cin>>a>>b;
		if(check(a)==b&&check(b)==a){
			cout<<"YES"<<endl;
		}else{
			cout<<"NO"<<endl;
		}
		}
		
	}
	
	return 0;
} 
